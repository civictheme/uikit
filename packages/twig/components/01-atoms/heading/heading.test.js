const template = 'components/01-atoms/heading/heading.twig';

describe('Heading Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      content: 'Sample Heading',
    });

    expect(c.querySelectorAll('.ct-heading')).toHaveLength(1);
    expect(c.querySelector('.ct-heading').textContent.trim()).toEqual('Sample Heading');
    expect(c.querySelector('.ct-heading').tagName.toLowerCase()).toEqual('h2'); // Default level is 2

    assertUniqueCssClasses(c);
  });

  test('renders with optional attributes', async () => {
    const c = await dom(template, {
      content: 'Sample Heading',
      level: '3',
      attributes: 'data-test="true"',
      modifier_class: 'custom-class',
      theme: 'dark',
    });

    expect(c.querySelectorAll('.ct-heading.custom-class.ct-theme-dark')).toHaveLength(1);
    expect(c.querySelector('.ct-heading').textContent.trim()).toEqual('Sample Heading');
    expect(c.querySelector('.ct-heading').tagName.toLowerCase()).toEqual('h3');
    expect(c.querySelector('.ct-heading').getAttribute('data-test')).toEqual('true');

    assertUniqueCssClasses(c);
  });

  test('does not render when content is empty', async () => {
    const c = await dom(template, {
      content: '',
    });

    expect(c.querySelectorAll('.ct-heading')).toHaveLength(0);
  });
});
