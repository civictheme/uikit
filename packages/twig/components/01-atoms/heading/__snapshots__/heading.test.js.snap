// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component does not render when content is empty 1`] = `
<div>
  



</div>
`;

exports[`Heading Component renders with optional attributes 1`] = `
<div>
  


  
  <h3
    class="ct-heading ct-theme-dark custom-class"
    data-test="true"
  >
    Sample Heading
  </h3>
  

</div>
`;

exports[`Heading Component renders with required attributes 1`] = `
<div>
  


  
  <h2
    class="ct-heading ct-theme-light "
  >
    Sample Heading
  </h2>
  

</div>
`;
